### PR TITLE
all functionality added

### PR DESCRIPTION
**Pull request by jaydipkmulani**
functionality added to sighn in button
and creted secure routinh

### CODE REVIEW

**Comment by Dz25** [comment]
Resolve merge conflicts and make sure it works as intended, no need for authentication on categories view, only profile view is good enough. Dont see search function implemented

File: src/components/Header.vue [hunk 1]
```diff
@@ -1,91 +1,152 @@
...
-})
+  window.addEventListener("scroll", handleScroll);
+});
+
 
 function handleScroll() {
-  isScrolled.value = window.pageYOffset > 0
+  isScrolled.value = window.pageYOffset > 0;
 }
 </script>
```
Comment (Dz25) [inline src/components/Header.vue:24]: Search function needed

File: src/components/Signin.vue [hunk 2]
```diff
@@ -1,134 +1,99 @@
...
-      border-bottom: 1px solid #ddd;
-      box-shadow: inset 0 1px 2px rgba(0, 0, 0, .4),
-        0 -1px 1px #fff,
-        0 1px 0 #fff;
-      overflow: hidden;
-      &:focus {
-        outline: none;
-        background-color: #fff;
+      if (response.status==200){
+        localStorage.setItem("user-info",JSON.stringify(response.data))
```
Comment (Dz25) [inline src/components/Signin.vue:71]: Only need to set userId

File: src/router/index.js [hunk 3]
```diff
@@ -1,33 +1,34 @@
...
-  ]
-})
+    { path: "/", name: "home", component: HomeView, meta: { requiresAuth: true } },
+    { path: "/pages/business", component: GeneralPagecard, props: { catagory: "business" }, meta: { requiresAuth: true } },
+    { path: "/pages/entertainment", component: GeneralPagecard, props: { catagory: "entertainment" }, meta: { requiresAuth: true } },
+    { path: "/pages/general", component: GeneralPagecard, props: { catagory: "general" }, meta: { requiresAuth: true } },
+    { path: "/pages/health", component: GeneralPagecard, props: { catagory: "health" }, meta: { requiresAuth: true } },
+    { path: '/pages/science', component: GeneralPagecard, props: { catagory: "science" }, meta: { requiresAuth: true } },
+    { path: "/pages/sports", component: GeneralPagecard, props: { catagory: "sports" }, meta: { requiresAuth: true } },
+    { path: "/pages/technology", component: GeneralPagecard, props: { catagory: "technology" }, meta: { requiresAuth: true } },
```
Comment (Dz25) [inline src/router/index.js:17]: No need to require auth on these page, people should be able to see these without sign in

File: src/components/Signin.vue (hunk 2 above)
Comment (jaydipkmulani) [reply src/components/Signin.vue:71]: userid is user-info.id

Comment (Dz25) [reply src/components/Signin.vue:71]: You need to use json.parse when access the data. You also leave the password for other to see when setting the whole object. You should set userId and username as key value pair only

**Pull request closed without merging**